Harden against accidental leaking of the password

maltesl opened the pull request:
In order to prevent submission of the password form of the
self-decrypting HTML page from causing the password from being sent over
the network, the JavaScript method `preventDefault()` was used.
However, this fails to take effect if the browser does not have
JavaScript enabled, and could ultimately lead to the password being
leaked to the network, which indeed happens with Lynx 2.8.9rel.1.

Hence set the action attribute explicitly to prevent the default action
from sending anything over the network. Also hide the form and display
a helpful message instead if the browser does not have JavaScript
enabled.

maltesl committed:
Harden against accidental leaking of the password

In order to prevent submission of the password form of the
self-decrypting HTML page from causing the password from being sent over
the network, the JavaScript method preventDefault() was used.
However, this fails to take effect if the browser does not have
JavaScript enabled, and could ultimately lead to the password being
leaked to the network, which indeed happens with Lynx 2.8.9rel.1.

Hence set the action attribute explicitly to prevent the default action
from sending anything over the network. Also hide the form and display
a helpful message instead if the browser does not have JavaScript
enabled.

fmeum commented:
Thanks for the contribution! While you are at it, could you also add the explicit action attribute and the `<noscript>` message to the encryption part of the page?

For future reference: According to the HTML spec, the values of form elements without a `name` attribute are not sent when the form is submitted, but Lynx behaves differently.

maltesl committed:
Harden against accidental leaking of information

Changes similar to those in commit db1e684, but this time for the
encryption page.

fmeum commented:
Thanks!
